clpy/core/core.pyx

```diff
@@ -4247,22 +4249,23 @@ def _nonzero_kernel(src_dtype, src_ndim, index_dtype, dst_dtype):
     dst_dtype = _get_typename(dst_dtype)
 
     source = string.Template("""
-        extern "C" __global__ void ${name}(const CArray<${src_dtype}, 1> src,
-            CIndexer<${src_ndim}> shape,
+        __kernel void ${name}(
+            const CArray<${src_dtype}, 1> src,
+            CIndexer_${src_ndim} shape,
             const CArray<${index_dtype}, 1> scaned_index,
             CArray<${dst_dtype}, 1> dst){
 
-            int thid = blockIdx.x * blockDim.x + threadIdx.x;
+            int thid = get_group_id(0) * get_local_size(0) + get_local_id(0);
```

Comment:
Why don't you use `get_global_id(0)`?

Reply:
I merely focused on porting the code.
`get_global_id` also works fine, so I modified this line.

Reply:
Porting is not transliteration, but translation. Please go OpenCL way!